src/Hosting/TestHost/src/ResponseBodyWriterStream.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.TestHost
+{
+    internal class ResponseBodyWriterStream : Stream
+    {
+        private readonly ResponseBodyPipeWriter _responseWriter;
+        private readonly Func<bool> _allowSynchronousIO;
+
+        public ResponseBodyWriterStream(ResponseBodyPipeWriter responseWriter, Func<bool> allowSynchronousIO)
```

Comment:
Why not have this ctor take the return value of `responseWriter.AsStream()` instead of copying all the `AsStream()` logic.  Turning the ResponseBodyWriterStream into a decorator will probably be less fragile since any fixes/improvements to the PipeWriterStream will get picked up automatically.

Reply:
Maybe if there were anything novel in PipeWriterStream, but there isn't. I still have to override just as many Stream APIs. https://github.com/dotnet/corefx/blob/master/src/System.IO.Pipelines/src/System/IO/Pipelines/PipeWriterStream.cs

Reply:
We can make it public, file an issue though.

Reply:
https://github.com/dotnet/corefx/issues/39224